hangman/lib/hangman/impl.ex

```diff
@@ -0,0 +1,91 @@
+defmodule Hangman.Impl do
+  # Implementation for Hangman Interface.
+
+  import Hangman.Helper
+
+  defmodule Tally do
```

Comment:
The idea behind the tally is that the game has two separate states: internal and external.

The internal state contains stuff the game module needs to run, and the external state, the `tally` is the values used by clients of this code.

This is where two bugs prevented me from running tests: the spelling of `initializing` and the absence of `last_guess`.

### works -8